### DISCUSSION TITLE
Link: Advancing the Search Frontier with AI Agents

### DISCUSSION THREAD

**Discussion by danieldekay** [assafelovic/gpt-researcher#804] [Ideas]
Just found the paper below. Might contain some great ideas that GPT-Researcher could use to become even better!

![image](https://github.com/user-attachments/assets/538abd1e-7970-429c-b6c6-c3f5a58b6a86)
https://cacm.acm.org/research/advancing-the-search-frontier-with-ai-agents/

**Comment by ElishaKay** [comment]
Sup @danieldekay 

For custom agents with custom tooling, I'm sensing a custom [Langgraph flow](https://www.langchain.com/langgraph) would be the safest bet.

P.S. if you're running on Mac, there is a [Langgraph Studio](https://studio.langchain.com/) desktop app which is excellent

**Reply by TomLucidor** [reply]
Are there other drag-and-drop tools as well for different pipelines? Or are they too inflexible?

**Reply by ElishaKay** [reply]
I've heard good things about these 2:

https://n8n.io

https://www.make.com
